RELEASING.md

```diff
@@ -0,0 +1,61 @@
+## Releasing Using a Long-Lived Version Branch Approach
+
+### Release Procedure
```

Comment:
question: As the `main` branch is always ready to be release., why not simply create a tag from there?

Long running release branches are a pain, as they tend to lead to more merging (overhead). 
When a bug is identified in a release tag, you can branch from that tag to fix, then tag that for the bugfix version.

Moto: create branches only when you really, really need to 👍

Reply:
Though I agree that long running release branches tend to more merging and conflicts for commonly updated files, e.g., changelog, I think it's cleaner to have release branches than bug-fix branches.

I don't think main is always for release, e.g., it may be mid-large-PR, mid-epic and/or multiple epics may be worked in parallel, so commits need to be cherry-picked for releases. 

My preference is have consistent existence of branches rather than creating them only when there's something to fix (the bug fix branch may then end up as long-lived). Having tags for a release on different branches makes it harder to find the latest tag for build scripts that include version info.

Reply:
The process for managing releases is one aspect of our development process / philosophy.

Compared to previous iteration of source control, due to how Git works, it can effectively support many more approaches in use.

> I don't think main is always for release

The consistent advice I'm finding (and agree with) is that `main` should always be releasable. Always working, never with partially implemented features (they should exist on feature branches, as per our [documented process}(https://github.com/windranger-io/solidity-project-template/blob/a5fdabf7a8560f3ba076cb58f66f80768fbeca45/docs/development_process.md) ).

> it may be mid-large-PR, mid-epic and/or multiple epics may be worked in parallel

Usually feature flags are used for this, or long running feature branches. 

> My preference is have consistent existence of branches rather than creating them only when there's something to fix (the bug fix branch may then end up as long-lived). Having tags for a release on different branches makes it harder to find the latest tag for build scripts that include version info.

The process I'm describing is different to that. I'm suggesting branching from a tag t(for bug fix), then releasing (tagging) the fix and deleting that branch.

It comes down to the point of branch existence. I lean towards the minimalist approach, that if it can be removed, then why not?
When a branch exists, I see them either as a maintenance cost (if they are maintained) or as noise (if they are not needed, this also why I favour forking rather than branching on the parent repo).

Reply:
I understand. I'm on board with your suggested approach. It's good to know that tagging keeps git history even after branch deletion, and achieves the result I was looking for. Thanks for the detailed and clear explanation, @CjHare.